models/restful-model-collection.coffee

```diff
@@ -77,10 +77,17 @@ class RestfulModelCollection
       , (chunkCallback) =>
         chunkOffset = offset + accumulated.length
         chunkLimit = Math.min(REQUEST_CHUNK_SIZE, limit - accumulated.length)
-        @getModelCollection(params, chunkOffset, chunkLimit).then (models) ->
-          accumulated = accumulated.concat(models)
-          finished = models.length < REQUEST_CHUNK_SIZE or accumulated.length >= limit
-          chunkCallback()
+        @getModelCollection(params, chunkOffset, chunkLimit)
+          .then((models) ->
+            accumulated = accumulated.concat(models)
+            finished = models.length < REQUEST_CHUNK_SIZE or accumulated.length >= limit
+            chunkCallback()
+          )
+          .catch((err) ->
+            if err
+              callback(err) if callback
```

Comment:
Instead of escaping out of this function immediately, it seems more expected for us to call `chunkCallback(err)` and let the async.until callback handle the error logic

Reply:
...in which case you could just pass the chunkCallback directly to the catch `catch(chunkCallback)`